README.md
# xtcbase

debian/changelog
xtcbase (1.8.0) UNRELEASED; urgency=low

 * xtc 3
 
 -- Stephan Eisler <stephan@eisler.de>  Thu, 11 Feb 2021 20:59:00 +0000

xtcbase (1.0.29) UNRELEASED; urgency=low

  * firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 19:16:00 +0000
 
xtcbase (1.0.28) UNRELEASED; urgency=low

  * firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 17:55:00 +0000
 
xtcbase (1.0.27) UNRELEASED; urgency=low

  * firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 16:46:00 +0000

xtcbase (1.0.26) UNRELEASED; urgency=low

  * firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 13:56:00 +0000

xtcbase (1.0.25) UNRELEASED; urgency=low

  * firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 12:34:00 +0000
 
xtcbase (1.0.24) UNRELEASED; urgency=low

  * firstboot testing

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 10:11:00 +0000
 
xtcbase (1.0.23) UNRELEASED; urgency=low

  * firstboot testing (early and late)

 -- Stephan Eisler <stephan@eisler.de>  Sat, 12 Jan 2021 09:00:00 +0000

xtcbase (1.0.22) UNRELEASED; urgency=low

  * firstboot testing (reboot)

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 21:09:00 +0000

xtcbase (1.0.21) UNRELEASED; urgency=low

  * firstboot testing

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 17:46:00 +0000
 
xtcbase (1.0.20) UNRELEASED; urgency=low

  * firstboot testing

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 16:57:00 +0000
 
xtcbase (1.0.19) UNRELEASED; urgency=low

  * reboot after firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 15:51:00 +0000
 
xtcbase (1.0.18) UNRELEASED; urgency=low

  * firstboot

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 14:57:00 +0000


xtcbase (1.0.17) UNRELEASED; urgency=low

  * fix partition

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 14:03:00 +0000
 
xtcbase (1.0.16) UNRELEASED; urgency=low

  * fix path

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 11:51:00 +0000

xtcbase (1.0.15) UNRELEASED; urgency=low

  * install via deb package postinst

 -- Stephan Eisler <stephan@eisler.de>  Sat, 09 Jan 2021 11:11:00 +0000


xtcbase (1.0.14) UNRELEASED; urgency=low

  * inital firstboot config

 -- Stephan Eisler <stephan@eisler.de>  Wed, 07 Jan 2021 17:18:00 +0000

xtcbase (1.0.13) UNRELEASED; urgency=low

  * add package xterm

 -- Stephan Eisler <stephan@eisler.de>  Wed, 07 Jan 2021 14:14:00 +0000

xtcbase (1.0.12) UNRELEASED; urgency=low

  * removed network dependency

 -- Stephan Eisler <stephan@eisler.de>  Wed, 07 Jan 2021 09:12:00 +0000

xtcbase (1.0.11) UNRELEASED; urgency=low

  * only testing

 -- Stephan Eisler <stephan@eisler.de>  Wed, 05 Jan 2021 08:21:00 +0000
 
xtcbase (1.0.10) UNRELEASED; urgency=low

  * only testing

 -- Stephan Eisler <stephan@eisler.de>  Wed, 04 Jan 2021 20:01:00 +0000
 
xtcbase (1.0.9) UNRELEASED; urgency=low

  * fix service x flags

 -- Stephan Eisler <stephan@eisler.de>  Wed, 04 Jan 2021 18:17:00 +0000
 
xtcbase (1.0.8) UNRELEASED; urgency=low

  * fix firstboot script

 -- Stephan Eisler <stephan@eisler.de>  Wed, 03 Jan 2021 17:01:00 +0000
 
xtcbase (1.0.7) UNRELEASED; urgency=low

  * firstboot script

 -- Stephan Eisler <stephan@eisler.de>  Wed, 03 Jan 2021 12:23:00 +0000
 
xtcbase (1.0.6) UNRELEASED; urgency=low

  * fix argon1 script path

 -- Stephan Eisler <stephan@eisler.de>  Wed, 02 Jan 2021 12:10:00 +0000
 
xtcbase (1.0.5) UNRELEASED; urgency=low

  * remove pi user

 -- Stephan Eisler <stephan@eisler.de>  Wed, 02 Jan 2021 12:10:00 +0000

xtcbase (1.0.4) UNRELEASED; urgency=low

  * fix openbox config

 -- Stephan Eisler <stephan@eisler.de>  Wed, 31 Dec 2020 15:07:00 +0000
 
xtcbase (1.0.3) UNRELEASED; urgency=low

  * removed base-clients dependency

 -- Stephan Eisler <stephan@eisler.de>  Wed, 31 Dec 2020 12:11:00 +0000
 
xtcbase (1.0.2) UNRELEASED; urgency=low

  * Sign package

 -- Stephan Eisler <stephan@eisler.de>  Wed, 31 Dec 2020 10:17:03 +0000
 
xtcbase (1.0.1) UNRELEASED; urgency=low

  * Initial release

 -- Stephan Eisler <stephan@eisler.de>  Wed, 23 Dec 2020 13:33:00 +0000


xtc/firstboot-early.sh
#!/bin/bash

SOURCE=/opt/thinclient/install
#rm $SOURCE/install.err 2> /dev/null
DIR=/home/thinclient
DESTINATION=/opt/thinclient
VAR=/var/thinclient
DATA=/data
echo "* start installation"
MODEL=""
OS="U"
#if (cat /etc/issue | grep -i "RASPBIAN")
#then
#  OS="R"
#  # detect pi hardware 
#  MODEL=$(cat /proc/cpuinfo |grep "Revision" | cut -d: -f2 | sed 's/^[ \t]*//')
#  if [ "$MODEL" = "a22082" ] || [ "$MODEL" = "a220a0" ] || [ "$MODEL" = "a32082" ] || [ "$MODEL" = "a52082" ] || [ "$MODEL" = "a22083" ] || [ "$MODEL" = "a02100" ] || [ "$MODEL" = "a020d3" ] || [ "$MODEL" = "a2082" ] || [ "$MODEL" = "a20a0" ]
#  then
#    MODEL="PI3"
#  else
#    MODEL="PI4"
#  fi
#fi

#echo "System:" $OS
#echo "Model:" $MODEL

#echo "16. create data directory"
#mkdir $DATA 2>> /dev/null

#echo "17. mount data directory"
#if [ "$OS" = "R" ]
#then
#  cat /etc/fstab $SOURCE/fstab_data > /tmp/fstab
#  mv /tmp/fstab /etc/fstab
#  mount -a
#fi

echo "18. config data directory"
chmod a=rwx $DATA 2>> /dev/null
mkdir $DATA/thinclient 2>> $SOURCE/install.err
chown thinclient $DATA/thinclient 2>> $SOURCE/install.err
chgrp thinclient $DATA/thinclient 2>> $SOURCE/install.err
mkdir $DATA/thinclient/config 2>> $SOURCE/install.err
chown thinclient $DATA/thinclient/config 2>> $SOURCE/install.err
chgrp thinclient $DATA/thinclient/config 2>> $SOURCE/install.err
mkdir $DATA/vpn 2>> $SOURCE/install.err
chown thinclient $DATA/vpn 2>> $SOURCE/install.err
chgrp thinclient $DATA/vpn 2>> $SOURCE/install.err
mkdir $DATA/usb 2>> $SOURCE/install.err
chown thinclient $DATA/usb 2>> $SOURCE/install.err
chgrp thinclient $DATA/usb 2>> $SOURCE/install.err


#echo "23. install driver for pi desktop case (Raspbery Pi 3 only)"
#if [ "$OS" = "R" ]
#then
#  dpkg -i /opt/thinclient/install/desktop/pidesktop-base.deb
#  echo " " >> /boot/config.txt
#  echo " " >> /boot/config.txt
#  cp $SOURCE/desktop/pidesktop.sh $DESTINATION/desktop
#  cp $SOURCE/desktop/pidesktop-powerkey.service /lib/systemd
#  cp $SOURCE/desktop/pidesktop-reboot.service /lib/systemd
#  cp $SOURCE/desktop/pidesktop-shutdown.service /lib/systemd
#  cp $SOURCE/desktop/pidesktop-powerkey.service /etc/systemd/system
#  cp $SOURCE/desktop/pidesktop-reboot.service //etc/systemd/system
#  cp $SOURCE/desktop/pidesktop-shutdown.service /etc/systemd/system
#  chown root:root $DESTINATION/desktop/pidesktop.sh
#  chown root:root /lib/systemd/pidesktop-powerkey.service 
#  chown root:root /lib/systemd/pidesktop-reboot.service
#  chown root:root /lib/systemd/pidesktop-shutdown.service
#  chown root:root /etc/systemd/system/pidesktop-powerkey.service 
#  chown root:root /etc/systemd/system/pidesktop-reboot.service
#  chown root:root /etc/systemd/system/pidesktop-shutdown.service
#fi

#echo "25. install software for argon1 case (Raspberry Pi 4 only)"
#if [ "$OS" = "R" ]
#then
  #$SOURCE/desktop/argon1.sh
  #systemctl disable argononed
#fi



xtc/firstboot.sh
#!/bin/bash

SOURCE=/opt/thinclient/install
rm $SOURCE/install.err 2> /dev/null
DIR=/home/thinclient
DESTINATION=/opt/thinclient
VAR=/var/thinclient
DATA=/data
echo "* start installation"
MODEL=""
OS="U"
if (cat /etc/issue | grep -i "RASPBIAN")
then
  OS="R"
  # detect pi hardware 
  MODEL=$(cat /proc/cpuinfo |grep "Revision" | cut -d: -f2 | sed 's/^[ \t]*//')
  if [ "$MODEL" = "a22082" ] || [ "$MODEL" = "a220a0" ] || [ "$MODEL" = "a32082" ] || [ "$MODEL" = "a52082" ] || [ "$MODEL" = "a22083" ] || [ "$MODEL" = "a02100" ] || [ "$MODEL" = "a020d3" ] || [ "$MODEL" = "a2082" ] || [ "$MODEL" = "a20a0" ]
  then
    MODEL="PI3"
  else
    MODEL="PI4"
  fi
fi

echo "System:" $OS
echo "Model:" $MODEL


echo "1. install packages"
# delete pulseaudio while problems with also sound
apt-get -y remove pulseaudio

echo "3. add user"
rmdir /home/thinclient/Desktop
rmdir /home/thinclient/Documents
rmdir /home/thinclient/Downloads
rmdir /home/thinclient/Music
rmdir /home/thinclient/Pictures
rmdir /home/thinclient/Public
rmdir /home/thinclient/Templates
rmdir /home/thinclient/Videos


echo "16. create data directory"
#mkdir $DATA 2>> /dev/null

echo "17. mount data directory"
#if [ "$OS" = "R" ]
#then
#  cat /etc/fstab $SOURCE/fstab_data > /tmp/fstab
#  mv /tmp/fstab /etc/fstab
#  mount -a
#fi

echo "18. config data directory"
chmod a=rwx $DATA 2>> /dev/null
mkdir $DATA/thinclient 2>> $SOURCE/install.err
chown thinclient $DATA/thinclient 2>> $SOURCE/install.err
chgrp thinclient $DATA/thinclient 2>> $SOURCE/install.err
mkdir $DATA/vpn 2>> $SOURCE/install.err
chown thinclient $DATA/vpn 2>> $SOURCE/install.err
chgrp thinclient $DATA/vpn 2>> $SOURCE/install.err
mkdir $DATA/usb 2>> $SOURCE/install.err
chown thinclient $DATA/usb 2>> $SOURCE/install.err
chgrp thinclient $DATA/usb 2>> $SOURCE/install.err

echo "19. config read only fs"
if [ "$OS" = "R" ]
then
  sed '/\/ /s/defaults/ro,defaults/g' -i /etc/fstab
  sed '/\/boot/s/defaults/ro,defaults/g' -i /etc/fstab
  cat /etc/fstab $SOURCE/fstab_ro > /tmp/fstab
  mv /tmp/fstab /etc/fstab
fi

#echo "23. install driver for pi desktop case (Raspbery Pi 3 only)"
#if [ "$OS" = "R" ]
#then
#  dpkg -i /opt/thinclient/install/desktop/pidesktop-base.deb
#  echo " " >> /boot/config.txt
#  echo " " >> /boot/config.txt
#  cp $SOURCE/desktop/pidesktop.sh $DESTINATION/desktop
#  cp $SOURCE/desktop/pidesktop-powerkey.service /lib/systemd
#  cp $SOURCE/desktop/pidesktop-reboot.service /lib/systemd
#  cp $SOURCE/desktop/pidesktop-shutdown.service /lib/systemd
#  cp $SOURCE/desktop/pidesktop-powerkey.service /etc/systemd/system
#  cp $SOURCE/desktop/pidesktop-reboot.service //etc/systemd/system
#  cp $SOURCE/desktop/pidesktop-shutdown.service /etc/systemd/system
#  chown root:root $DESTINATION/desktop/pidesktop.sh
#  chown root:root /lib/systemd/pidesktop-powerkey.service 
#  chown root:root /lib/systemd/pidesktop-reboot.service
#  chown root:root /lib/systemd/pidesktop-shutdown.service
#  chown root:root /etc/systemd/system/pidesktop-powerkey.service 
#  chown root:root /etc/systemd/system/pidesktop-reboot.service
#  chown root:root /etc/systemd/system/pidesktop-shutdown.service
#fi

#echo "25. install software for argon1 case (Raspberry Pi 4 only)"
#if [ "$OS" = "R" ]
#then
  #$SOURCE/desktop/argon1.sh
  #systemctl disable argononed
#fi

